### PR TITLE
Switch template tests to tape with full coverage & tweaks.

### PR DESCRIPTION
**Pull request by bucaran**
+ Tape tests with coverage report and babel
+ Lint with ESLint
+ Lint on precommit
+ Remove semicolons
+ Add default URL based in user's GitHub by default
+ Simplify generator's test
+ Small tweaks to README

I'm sorry it took me an extra day to pull this off, but there it is. There may be some changes you don't agree with, but you may still benefit from my investigation regarding coverage for tape + Babel.

Let me know if this is alright.

### CODE REVIEW

**Comment by iamstarkov** [comment]
yay, great

**Comment by iamstarkov** [comment]
will take a closer look today

**Comment by bucaran** [comment]
:smile: Cool

File: app/index.js [hunk 1]
```diff
@@ -1,86 +1,103 @@
...
-      cpTpl('gitignore',     '.gitignore');
-      cpTpl('npmignore',     '.npmignore');
-      cpTpl('travis.yml',    '.travis.yml');
+      this.template('package.json')
+      this.template('README.md')
+      if (this.hasChangelog) {
+        this.template("CHANGELOG.md")
+      }
+      this.template('index.js',      'src/index.js')
+      this.template('test.js',       'test/index.js')
```
Comment (iamstarkov) [inline app/index.js]: if you dont mind i prefer to have module’s structure as flat as possible, so can you keep this thing in previous state

File: app/index.js [hunk 2]
```diff
@@ -1,86 +1,103 @@
...
+      this.template('editorconfig',  '.editorconfig')
+      this.template('gitignore',     '.gitignore')
+      this.template('npmignore',     '.npmignore')
+      this.template('travis.yml',    '.travis.yml')
+      this.template('eslintrc',      '.eslintrc')
 
-      cb();
-    }.bind(this));
+      done()
```
Comment (iamstarkov) [inline app/index.js]: 👍

File: app/templates/README.md [hunk 3]
```diff
@@ -16,7 +16,7 @@
 ```js
 import <%= camelModuleName %> from '<%= moduleName %>';
 
-<%= camelModuleName %>('unicorns'); // unicorns
+<%= camelModuleName %>(true); // true
```
Comment (iamstarkov) [inline app/templates/README.md]: you dont love unicorns?

File: app/templates/index.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export default function (input) {
```
Comment (iamstarkov) [inline app/templates/index.js]: named functions are better for debug, that’s why I switched to them from anonymous https://github.com/iamstarkov/generator-tiny-es-nm/commit/e7fc5e2a98f7cca9e39122df01b0cf904e16a261

Comment (iamstarkov) [inline app/templates/index.js]: so, pls, revert it =)

File: app/templates/package.json [hunk 5]
```diff
@@ -2,19 +2,25 @@
   "name": "<%= moduleName %>",
   "version": "<%= moduleVersion %>",
   "description": "<%= moduleDesc %>",
-  "main": "index.es5.js",
+  "main": "dist/index.js",
+  "files": [
+    "bin/",
+    "dist/",
+    "LICENSE"
+  ],
```
Comment (iamstarkov) [inline app/templates/package.json]: License already included in this list https://docs.npmjs.com/files/package.json

File: app/templates/README.md (hunk 3 above)
Comment (bucaran) [inline app/templates/README.md]: :laughing: I've never seen on.

File: app/templates/index.js (hunk 4 above)
Comment (bucaran) [inline app/templates/index.js]: > named functions are better for debug,

You are right.

File: app/templates/package.json (hunk 5 above)
Comment (bucaran) [inline app/templates/package.json]: You don't like keeping a LICENSE file?

File: app/templates/package.json [hunk 6]
```diff
@@ -2,19 +2,25 @@
...
   "scripts": {
-    "coverage": "isparta cover _mocha index.js -- --require babel/register",
-    "precoveralls": "npm run coverage",
-    "coveralls": "coveralls < coverage/lcov.info",
-    "test": "mocha --require babel/register",
+    "clean": "rimraf dist",
+    "lint": "eslint src test",
+    "build": "npm run clean && npm run lint && babel --optional runtime src -d dist",
+    "test": "npm run build && babel-node test/*.js",
```
Comment (iamstarkov) [inline app/templates/package.json]: why do I need to build project, if I can test original es6 version? like before

Comment (bucaran) [inline app/templates/package.json]: It's a plumbing script, but after working on these for a few days and also today I came up with an improved workflow which I published in my generator. I think we can close this PR. But I am pretty sure you can benefit from my research on using tape with coveralls and ES6. It seems simple now, but that was a good chunk of my weekend :laughing:

File: app/templates/package.json [hunk 7]
```diff
@@ -2,19 +2,25 @@
   "name": "<%= moduleName %>",
```
Comment (iamstarkov) [inline app/templates/package.json]: about renaming, does "package.json" working properly? when I tried it, it failed

File: app/templates/README.md (hunk 3 above)
Comment (iamstarkov) [inline app/templates/README.md]: nobody has =)

File: app/templates/package.json (hunk 7 above)
Comment (bucaran) [inline app/templates/package.json]: Sorry, that damn `,`. I kept making the same mistake over and over every time I removed a field or changed the JSON a little bit.

File: app/templates/package.json (hunk 5 above)
Comment (iamstarkov) [inline app/templates/package.json]: no, you get it wrong, checkout docs link above:

> Certain files are always included, regardless of settings:

> * package.json
> * README (and its variants)
> * CHANGELOG (and its variants)
> * LICENSE / LICENCE

File: app/templates/package.json (hunk 6 above)
Comment (iamstarkov) [inline app/templates/package.json]: > I think we can close this PR

okay

> I came up with an improved workflow which I published in my generator. 

congrats 💥

File: app/templates/package.json (hunk 5 above)
Comment (bucaran) [inline app/templates/package.json]: I just learned that! :+1: Yay

File: app/templates/package.json (hunk 6 above)
Comment (bucaran) [inline app/templates/package.json]: The more I think about it, the problem with npm scripts is that while on one hand they are simple to reason about, relatively short and all affordably written in your package.json (which is really nice), they are very tightly couples, it's like a micro DSL for your own app. If you can memorize your preferred script workflow you'll be okay for a while, but it's not really flexible.

I am thinking of adding an option to the other generator to scaffold out a project with a Flyfile, maybe later when fly 1.0 lands early next month with the complete feature set and a stable API.

Comment (iamstarkov) [inline app/templates/package.json]: I have better idea, will show it later

Comment (bucaran) [inline app/templates/package.json]: Thanks :+1:

**Comment by iamstarkov** [comment]
btw, have you seen [ava][ava] project?

[ava]: https://github.com/sindresorhus/ava

**Comment by bucaran** [comment]
I have, but didn't find it so relevant.

**Pull request closed without merging**